plugins/package-managers/conan/src/main/kotlin/Conan.kt

```diff
@@ -415,7 +416,7 @@ class Conan(
             val url = artifactEntry["url"].let { urlNode ->
                 (urlNode.takeIf { it.isTextual } ?: urlNode.first()).textValueOrEmpty()
             }
-            val hash = Hash.create(artifactEntry["sha256"].textValueOrEmpty())
+            val hash = Hash.create(artifactEntry["sha256"].textValueOrEmpty(), HashAlgorithm.SHA256.name)
```

Comment:
Yet better: `Hash(artifactEntry["sha256"].textValueOrEmpty(), HashAlgorithm.SHA256)`

Reply:
No, I didn't. Yet another drawback of us using data classes that get serialized also as interfaces 😞

But I wonder whether it matters: Serialization now always happens lower-case, and comparison already is case-insensitve. So maybe we could even consider dropping your second commit instead? Or at least document in its message that is only as consistency reason, but is not strictly required.

Reply:
If you compare the `Hash` instance with the `equals` function, then my second commit is required. 
So, it does have an effect IMO. What do you think?

Reply:
Makes sense. Then please remind us about that and add to your second commit's message that lowercasing is not required for verification, but for equality checks of instances.

My "Yet better" can be disregarded then, or changed to `Hash(artifactEntry["sha256"].textValueOrEmpty().lowercase(), HashAlgorithm.SHA256)`.